### PR TITLE
Read Route53 credentials from Vault

### PR DESCRIPTION
**Pull request by bonifaido**

### CODE REVIEW

File: cluster/hooks.go [hunk 1]
```diff
@@ -305,12 +309,26 @@ func RegisterDomainPostHook(input interface{}) error {
...
+		log.Errorf("Failed to read AWS credentials from Vault: %s", err.Error())
+		return err
+	}
+	if secret == nil {
+		log.Infoln("No AWS credentials for Route53 provided in Vault, exiting as this functionality is not enabled")
+		return nil
+	}
+
+	awsCredentials := cast.ToStringMapString(secret.Data["data"])
+	region := awsCredentials["region"]
```
Comment (stoader) [inline cluster/hooks.go]: Define the key as constant

File: cluster/hooks.go [hunk 2]
```diff
@@ -305,12 +309,26 @@ func RegisterDomainPostHook(input interface{}) error {
...
+	awsSecretId := awsCredentials["accessKeyId"]
```
Comment (stoader) [inline cluster/hooks.go]: I believe for this one we already have a constant defined

File: cluster/hooks.go [hunk 3]
```diff
@@ -305,12 +309,26 @@ func RegisterDomainPostHook(input interface{}) error {
 		return errors.Errorf("Wrong parameter type: %T", commonCluster)
 	}
 
-	region := ""       // TODO: this should come from config or vault
-	awsSecretId := ""  // TODO: this should come from vault
-	awsSecretKey := "" // TODO: this should come form vault
+	// This is how the secrets are expected to be written in Vault:
+	// vault kv put secret/banzaicloud/aws region=... accessKeyId=... secretAccessKey=...
```
Comment (stoader) [inline cluster/hooks.go]: Add this to dev docs as well

**Comment by bonifaido** [comment]
Docs and constants are updated now.

**Comment by bonifaido** [comment]
I'm fixing the test.

**Comment by bonifaido** [comment]
Tests fixed.

**Pull request merged**